src/components/kubectl/logs.ts

```diff
@@ -0,0 +1,184 @@
+    if (!podSummary) {
+        vscode.window.showErrorMessage('Can\'t find a pod!');
+        return;
+    }
+
+    const container = await selectContainerForPod(podSummary);
+    if (!container) {
+        return;
+    }
+
+    getLogsForContainer(kubectl, podSummary, container.name, displayMode);
+}
+
+/**
+ * Gets the logs for a container in a provided pod, in a provided namespace, in a provided container.
+ * @param kubectl
+ * @param podSummary
+ * @param containerName
+ * @param displayMode
+ */
+function getLogsForContainer(
+    kubectl: Kubectl,
+    podSummary: PodSummary,
+    containerName: string | undefined,
+    displayMode: LogsDisplayMode
+) {
+    let cmd = `logs ${podSummary.name}`;
+
+    if (podSummary.namespace) {
+        cmd = `${cmd} --namespace=${podSummary.namespace}`;
+    }
+
+    if (containerName) {
+        cmd = `${cmd} --container=${containerName}`;
+    }
+
+    if (displayMode === LogsDisplayMode.Follow) {
+        cmd = `${cmd} -f`;
+        kubectl.invokeInNewTerminal(cmd, FOLLOW_LOGS_TERMINAL);
+        return;
+    }
+
+    kubectl.invokeInSharedTerminal(cmd);
+}
+
+/**
+ * Searches for a pod yaml spec from the open document
+ * or from the currently selected namespace.
+ * @param kubectl
+ * @param displayMode
+ */
+async function findPod(kubectl: Kubectl, displayMode: LogsDisplayMode) {
+    const editor = vscode.window.activeTextEditor;
+
+    if (editor) {
+        return findPodFromOpenDocument(kubectl, editor, displayMode);
+    }
+
+    return await findPodFromCurrentNamespace(kubectl, displayMode);
+}
+
+/**
+ * Finds a Pod from the open editor.
+ * @param editor
+ * @param editor
+ * @param displayMode
+ */
+async function findPodFromOpenDocument(kubectl: Kubectl, editor: vscode.TextEditor, displayMode: LogsDisplayMode) {
+    const text = editor.document.getText();
+    try {
+        const obj: {} = yaml.safeLoad(text);
+        if (!isPod(obj)) {
+            return;
+        }
+
+        const podSummary = {
+            name: obj.metadata.name,
+            namespace: obj.metadata.namespace
+        };
+
+        getLogsForPod(kubectl, podSummary, displayMode);
+
+        return;
+    } catch (ex) {
+        host.showErrorMessage("Could not retrieve Pod from Active Document");
+    }
+}
+
+/**
+ * Alerts the user on pods available in the current namespace.
+ * @param kubectl
+ * @param displayMode
+ */
+async function findPodFromCurrentNamespace(kubectl: Kubectl, displayMode: LogsDisplayMode) {
+    const namespace = await kubectlUtils.currentNamespace(kubectl);
+
+    quickPickKindName(
+        [kuberesources.allKinds.pod],
+        { nameOptional: false },
+        (pod) => {
+
+            const podSummary: PodSummary = {
+                name: pod.split('/')[1],
+                namespace: namespace // should figure out how to handle namespaces.
+            };
+
+            getLogsForPod(kubectl, podSummary, displayMode);
```

Comment:
Same comment about `findPod` function naming.  In the old code, `findPod` took a callback by the look of it; that callback now seems to have been hardwired into the function definitions.

Reply:
👍